builtin/logical/rabbitmq/path_config_connection.go

```diff
@@ -0,0 +1,120 @@
+package rabbitmq
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+	"github.com/michaelklishin/rabbit-hole"
+)
+
+func pathConfigConnection(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/connection",
+		Fields: map[string]*framework.FieldSchema{
+			"connection_uri": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "RabbitMQ Management URI",
+			},
+			"username": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Username of a RabbitMQ management administrator",
+			},
+			"password": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "Password of the provided RabbitMQ management user",
+			},
+			"verify_connection": &framework.FieldSchema{
+				Type:        framework.TypeBool,
+				Default:     true,
+				Description: `If set, connection_uri is verified by actually connecting to the RabbitMQ management API`,
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathConnectionWrite,
+		},
+
+		HelpSynopsis:    pathConfigConnectionHelpSyn,
+		HelpDescription: pathConfigConnectionHelpDesc,
+	}
+}
+
+func (b *backend) pathConnectionWrite(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	uri := data.Get("connection_uri").(string)
+	username := data.Get("username").(string)
+	password := data.Get("password").(string)
+
+	if uri == "" {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"'connection_uri' is a required parameter.")), nil
+	}
+
+	if username == "" {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"'username' is a required parameter.")), nil
+	}
+
+	if password == "" {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"'password' is a required parameter.")), nil
+	}
+
+	// Don't check the connection_url if verification is disabled
+	verifyConnection := data.Get("verify_connection").(bool)
+	if verifyConnection {
+		// Create RabbitMQ management client
+		client, err := rabbithole.NewClient(uri, username, password)
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf(
+				"Error  info: %s", err)), nil
+		}
+
+		// Verify provided user is able to list users
+		_, err = client.ListUsers()
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf(
+				"Error validating connection info by listing users: %s", err)), nil
+		}
```

Comment:
A client can have more than one connection associated with it. I guess my thinking is, if the client has changed, *all* connections for that client should be closed, and the client re-opened. Or in the case of verifying, the connections should all be closed and the client torn down.

What we want to avoid is connections not closing properly and causing leaked file descriptors. This has hit us with a number of other backends due to connections not being closed properly.